erizo_controller/erizoController/models/Client.js

```diff
@@ -26,6 +26,20 @@ function listenToSocketEvents(client) {
   client.channel.on('disconnect', client.onDisconnect.bind(client));
 }
 
+function stopListeningToSocketEvents(client) {
+  log.info('Stop listening to socket events');
+  client.channel.socketRemoveListener('sendDataStream', client.onSendDataStream.bind(client));
```

Comment:
I think the `removeListener` call will not remove anything here.
`bind()` returns a new function every time it is called, so this function will be different from the one that was registered.

Reply:
You have to store the generated function somewhere if you want to remove it later:
```js
const handler = instance.onEvent.bind(instance);
on('some event', handler);
removeListener('some event', handler);
```

Reply:
Note that we might also just call `removeAllListeners` here instead.

Reply:
Good catch, thats not good at all. I will work on a fix later today.
Regarding `removeAllListeners` I prefer explicitly removing the listeners that we set up in `listenToSocketEvents`.